lib/protos.js

```diff
@@ -109,30 +125,60 @@ exports.track = function(track){};
  * // => []
  */
 
-exports.map = function(events, event){
-  var normalizedEvent = normalize(event);
-
-  return foldl(function(matchingEvents, val, key, events) {
-    // If true, this is a `mixed` value, which is structured like so:
-    //     { key: 'testEvent', value: { event: 'testEvent', someValue: 'xyz' } }
-    // We need to extract the key, which we use to match against
-    // `normalizedEvent`, and return `value` as part of `matchingEvents` if that
-    // match succeds.
-    if (type(events) === 'array') {
-      // If there's no key attached to this event mapping (unusual), skip this
-      // item.
-      if (!val.key) return matchingEvents;
-      // Extract the key and value from the `mixed` object.
-      key = val.key;
-      val = val.value;
+exports.map = function(mapping, key){
```

Comment:
Ah, I see—because this isn't necessarily a mapping of `events`. In that case I like your `options` naming up in the JSDoc block; let's rename everything to follow that convention. `key` doesn't fit in the case of array but I can't think of a better name for it so let's just go with that for now